connectors/okhttp/src/main/java/org/openstack4j/connectors/okhttp/HttpCommand.java

```diff
@@ -101,15 +110,23 @@ private void initialize() {
     /**
      * Create ConnectionPool optimized for short lived client with little chance to reuse connections.
      */
-    private ConnectionPool getConnectionPool() {
+    private static OkHttpClient getClient(Config config) {
         int maxIdleConnections = 0;
         // OkHttp creates "OkHttp ConnectionPool" thread per every ConnectionPool created to mange its connections. It
         // lives as long as the last connection made through it + its keep alive timeout. By default that it 5 min which
         // makes no sense for openstack4j since the connections or pools are not reused (after HttpCommand completion,
         // at least). Setting strict keepAlive duration here so the connections and threads does not hang around longer
         // than necessary.
         int keepAliveDuration = 500;
-        return new ConnectionPool(maxIdleConnections, keepAliveDuration, TimeUnit.MILLISECONDS);
+        ConnectionPool cp = new ConnectionPool(maxIdleConnections, keepAliveDuration, TimeUnit.MILLISECONDS);
```

Comment:
Author of the original workaround here. I like the approach of rectifying the misuse of OkHttp client, though I belive we have to rethink the keepalive duration and the size of the pool. The only reason this was ever tempered with was the numerous `OkHttpClient` instances took minutes to purge the threads and connections causing resource waste. I guess we can get back to OkHttp's defaults here as we are using the (single) client the was it was meant to be.

Reply:
I do not think we need any of this now: https://github.com/ContainX/openstack4j/commit/dc97e6733a0d1151cce69dd9f453a6b501faf323#diff-e08fdcb99d047ca41bbe3fe37e41802f